### PR TITLE
conversion macro for run2 digits

### PR DESCRIPTION
**Pull request by jbarrella**
This macro will allow the user to convert either raw data or run2 digits into run3 digits

### CODE REVIEW

File: Detectors/TRD/macros/convertRun2ToRun3Digits.C [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+      TTree* tr = (TTree*)run2DigitsFile.Get(Form("%s/TreeD", obj->GetName()));
+
+      for (int det = 0; det < 540; det++) {
+        digitMan->ClearArrays(det);
+        digitMan->ClearIndexes(det);
+      }
+
+      digitMan->ReadDigits(tr);
+
+      for (int det = 0; det < 540; det++) {
```
Comment (bazinski) [inline Detectors/TRD/macros/convertRun2ToRun3Digits.C:157]: You could replace 540 with AliTRDCommonParam::kNdet and a few other places where looping over 540

File: Detectors/TRD/macros/convertRun2ToRun3Digits.C [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+        digitMan->GetDigits(det)->Expand();
+
+        AliTRDarrayADC* digits;
+        digits = digitMan->GetDigits(det);
+        digits->Expand();
+
+        // TODO: check actual number of rows, from geometry
+        // here: play it safe, assume 12 rows everywhere
+        int nrows = 16;
```
Comment (bazinski) [inline Detectors/TRD/macros/convertRun2ToRun3Digits.C:172]: AliTRDfeeParam::fgNrowC1=16

File: Detectors/TRD/macros/convertRun2ToRun3Digits.C [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        if (stack == 2) {
+          nrows = 12;
```
Comment (bazinski) [inline Detectors/TRD/macros/convertRun2ToRun3Digits.C:174]: AliTRDfeeParam::fgkNrowC0=12

File: Detectors/TRD/macros/convertRun2ToRun3Digits.C [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        }
+
+        for (int row = 0; row < nrows; row++) {
+          for (int col = 0; col < 144; col++) {
```
Comment (bazinski) [inline Detectors/TRD/macros/convertRun2ToRun3Digits.C:178]: 144=AliTRDfeeParam::fgkNcol

File: Detectors/TRD/macros/convertRun2ToRun3Digits.C (hunk 2 above)
Comment (jbarrella) [reply Detectors/TRD/macros/convertRun2ToRun3Digits.C]: Had to use "get" methods to retrieve protected members in the AliTRDfeeParam class

Comment (bazinski) [reply Detectors/TRD/macros/convertRun2ToRun3Digits.C]: Ah right in O2 they are static constexpr. Not to mention I got the name wrong.

File: Detectors/TRD/macros/convertRun2ToRun3Digits.C [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+            // cout << "timebins: " << digitMan->GetDigits(det)->GetNtime() << endl;
+
+            for (int timebin = 0; timebin < digitMan->GetDigits(det)->GetNtime(); timebin++) {
+              int adc = digitMan->GetDigitAmp(row, col, timebin, det);
+
+              if (digits->GetNtime() > 30) {
+                cout << "----!!! --- number of times is greater than 30" << endl;
+              }
+
+              adctimes[timebin] = digits->GetData(row, col, timebin);
```
[the remark this answers is not in the document]
Comment (jbarrella) [reply Detectors/TRD/macros/convertRun2ToRun3Digits.C]: Thanks for the suggestion. Yes, I checked this and it seems that `digits->GetData(row, col, timebin)` is always equal to `digitMan->GetDigitAmp(row, col, timebin, det)` which makes sense given that "`digits`" comes from `digits = digitMan->GetDigits(det)`. So I changed the line to `adctimes[timebin] = adc` and did away with the "`digits`" object altogether.

[the remark this answers is not in the document]
Comment (jolopezl) [reply Detectors/TRD/macros/convertRun2ToRun3Digits.C]: Thanks for the clarification.

**Comment by jbarrella** [comment]
> I would remove all commented headers, or add a comment on why you would like to keep them like that.

This should also be resolved in the latest commit

**Pull request merged**